chainio/clients/wallet/fireblocks_wallet.go

```diff
@@ -191,21 +219,50 @@ func (t *fireblocksWallet) SendTransaction(ctx context.Context, tx *types.Transa
 		feeLevel = fireblocks.FeeLevelHigh
 	}
 
-	req := fireblocks.NewContractCallRequest(
-		"", // externalTxID
-		assetID,
-		account.ID,                // source account ID
-		contract.ID,               // destination account ID
-		tx.Value().String(),       // amount
-		hexutil.Encode(tx.Data()), // calldata
-		replaceTxByHash,           // replaceTxByHash
-		gasPrice,
-		gasLimit,
-		maxFee,
-		priorityFee,
-		feeLevel,
-	)
-	res, err := t.fireblocksClient.ContractCall(ctx, req)
+	var res *fireblocks.TransactionResponse
+	if len(tx.Data()) == 0 && tx.Value().Cmp(big.NewInt(0)) > 0 {
+		targetAccount, clientErr := t.getWhitelistedAccount(ctx, *tx.To())
+		if clientErr != nil {
+			return "", fmt.Errorf("error getting whitelisted account %s: %w", tx.To().Hex(), clientErr)
+		}
+		req := fireblocks.NewTransferRequest(
+			"", // externalTxID
+			assetID,
+			account.ID,                      // source account ID
+			targetAccount.ID,                // destination account ID
+			weiToEther(tx.Value()).String(), // amount in ETH
+			replaceTxByHash,                 // replaceTxByHash
+			gasPrice,
+			gasLimit,
+			maxFee,
+			priorityFee,
+			feeLevel,
+		)
+		res, err = t.fireblocksClient.Transfer(ctx, req)
+	} else if len(tx.Data()) > 0 {
+		contract, clientErr := t.getWhitelistedContract(ctx, *tx.To())
+		if clientErr != nil {
+			return "", fmt.Errorf("error getting whitelisted contract %s: %w", tx.To().Hex(), clientErr)
+		}
+		req := fireblocks.NewContractCallRequest(
+			"", // externalTxID
+			assetID,
+			account.ID,                      // source account ID
+			contract.ID,                     // destination account ID
+			weiToEther(tx.Value()).String(), // amount
+			hexutil.Encode(tx.Data()),       // calldata
+			replaceTxByHash,                 // replaceTxByHash
+			gasPrice,
+			gasLimit,
+			maxFee,
+			priorityFee,
+			feeLevel,
+		)
+		res, err = t.fireblocksClient.ContractCall(ctx, req)
+	} else {
+		return "", errors.New("transaction has no value and no data")
+	}
+
 	if err != nil {
 		return "", fmt.Errorf("error calling contract %s: %w", tx.To().Hex(), err)
```

Comment:
nit: this should prob read "error sending a transaction" now?

Reply:
good call